pintracker/maptracker/maptracker_test.go

```diff
@@ -320,3 +320,182 @@ func TestSyncAll(t *testing.T) {
 		t.Fatal("should have synced h2")
 	}
 }
+
+func TestUntrackTrack(t *testing.T) {
+	mpt := testMapPinTracker(t)
+	defer mpt.Shutdown()
+
+	h1, _ := cid.Decode(test.TestCid1)
+
+	// LocalPin
+	c := api.Pin{
+		Cid:                  h1,
+		Allocations:          []peer.ID{},
+		ReplicationFactorMin: -1,
+		ReplicationFactorMax: -1,
+	}
+
+	err := mpt.Track(c)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(time.Second / 2)
+
+	err = mpt.Untrack(h1)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestTrackUntrackWithCancel(t *testing.T) {
+	mpt := testMapPinTracker(t)
+	defer mpt.Shutdown()
+
+	h1, _ := cid.Decode(test.TestCid1)
+
+	// LocalPin
+	c := api.Pin{
+		Cid:                  h1,
+		Allocations:          []peer.ID{},
+		ReplicationFactorMin: -1,
+		ReplicationFactorMax: -1,
+	}
+
+	err := mpt.Track(c)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for {
+		opc, ok := mpt.optracker.get(c.Cid)
+		if !ok {
+			t.Fatal()
+		}
+		if opc.phase == phaseInProgress {
+			err = mpt.Untrack(h1)
```

Comment:
Once you untrack, it would be good to find a way to check that the operation was cancelled. I'm not sure about the best approach though. Probably the mock RPC can report whether the context in the operation was cancelled.